### PR TITLE
Enhance logging in masscanned

### PR DESCRIPTION
**Pull request by Frky**
Addresses Issue #3.

### CODE REVIEW

**Comment by Frky** [comment]
I take any feedback, this is a draft PR with an idea for logging that can be improved/restructured/revoked.

**Comment by Frky** [comment]
Example of `ConsoleLogger` output:

```
$ sudo ./target/debug/masscanned -vvvv --iface vethb 2>/dev/null
arp::init
arp::recv       who-has c6:e1:9d:72:2e:e8       00:00:00:00:00:00       10.255.255.3
arp::send       is-at   c6:23:de:89:7a:7f       10.255.255.3    c6:e1:9d:72:2e:e8       10.255.255.2
```

**Comment by p-l-** [comment]
Can you rebase against current master? Would make it easier to review.

**Comment by Frky** [comment]
Yes, that has just been done.

File: src/utils/loggers.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+impl Logger for ConsoleLogger {
+    fn init(&self) {
+        println!("arp::init");
+    }
+    fn arp_enabled(&self) -> bool {
+        self.arp
+    }
+    fn eth_enabled(&self) -> bool {
+        self.arp
```
Comment (p-l-) [inline src/utils/loggers.rs:67]: Is this intended?

Comment (Frky) [reply src/utils/loggers.rs:67]: Not intended. Thanks.

**Comment by Frky** [comment]
Ready to review (and to merge if OK). Minimalistic migration, some work is still to be done for proto dissectors.

New example of console log output:

 ```
# ./target/debug/masscanned --iface vethb
1644658461.52   arp     init
1644658461.52   eth     init
1644658461.52   ipv4    init
1644658461.52   ipv6    init
1644658461.52   icmpv4  init
1644658461.52   icmpv6  init
1644658461.52   tcp     init
1644658461.52   udp     init
1644658464.226  eth     recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f                                               Ipv4
1644658464.226  ipv4    recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1                               Icmp
1644658464.226  icmpv4  recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(8)     IcmpCode(0)
1644658464.226  icmpv4  send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(0)     IcmpCode(0)
1644658464.226  ipv4    send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Icmp
1644658464.226  eth     send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Ipv4
1644658465.239  eth     recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f                                               Ipv4
1644658465.240  ipv4    recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1                               Icmp
1644658465.240  icmpv4  recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(8)     IcmpCode(0)
1644658465.240  icmpv4  send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(0)     IcmpCode(0)
1644658465.240  ipv4    send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Icmp
1644658465.240  eth     send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Ipv4
1644658466.264  eth     recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f                                               Ipv4
1644658466.264  ipv4    recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1                               Icmp
1644658466.264  icmpv4  recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(8)     IcmpCode(0)
1644658466.264  icmpv4  send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(0)     IcmpCode(0)
1644658466.264  ipv4    send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Icmp
1644658466.265  eth     send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Ipv4
1644658467.265  eth     recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f                                               Ipv4
1644658467.265  ipv4    recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1                               Icmp
1644658467.265  icmpv4  recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(8)     IcmpCode(0)
1644658467.265  icmpv4  send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(0)     IcmpCode(0)
1644658467.265  ipv4    send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Icmp
1644658467.266  eth     send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Ipv4
1644658468.280  eth     recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f                                               Ipv4
1644658468.280  ipv4    recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1                               Icmp
1644658468.280  icmpv4  recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(8)     IcmpCode(0)
1644658468.280  icmpv4  send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(0)     IcmpCode(0)
1644658468.280  ipv4    send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Icmp
1644658468.280  eth     send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Ipv4
1644658469.281  eth     recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f                                               Ipv4
1644658469.281  ipv4    recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1                               Icmp
1644658469.281  icmpv4  recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(8)     IcmpCode(0)
1644658469.281  icmpv4  send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    IcmpType(0)     IcmpCode(0)
1644658469.281  ipv4    send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Icmp
1644658469.281  eth     send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       Icmp                    Ipv4
1644658469.436  eth     recv    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f                                               Arp
1644658469.436  arp     recv    c6:e1:9d:72:2e:e8       00:00:00:00:00:00       192.0.0.0       192.0.0.1       ArpOperation(1)
1644658469.436  arp     recv    c6:e1:9d:72:2e:e8       00:00:00:00:00:00       192.0.0.0       192.0.0.1       ArpOperation(1)
1644658469.436  arp     send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       ArpOperation(2)
1644658469.436  arp     send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f       192.0.0.0       192.0.0.1       ArpOperation(2)
1644658469.436  eth     send    c6:e1:9d:72:2e:e8       d6:19:4e:32:55:4f                                               Arp
```

**Pull request merged**